tests/common/devices.py

```diff
@@ -1262,6 +1279,48 @@ def get_crm_facts(self):
 
         return crm_facts
 
+    def stop_service(self, service_name, docker_name):
+        logging.debug("Stopping {}".format(service_name))
+        if self.is_service_fully_started(docker_name):
+            self.command("systemctl stop {}".format(service_name))
+        logging.debug("Stopped {}".format(service_name))
+
+    def delete_container(self, service):
+        if self.is_container_present(service):
+            self.command("docker rm {}".format(service))
+
+    def is_bgp_state_idle(self):
+        bgp_summary = self.command("show ip bgp summary")["stdout_lines"]
+
+        idle_count = 0
+        expected_idle_count = 0
+        for line in bgp_summary:
+            if "Idle (Admin)" in line:
+                idle_count += 1
+
+            if "Total number of neighbors" in line:
+                tokens = line.split()
+                expected_idle_count = int(tokens[-1])
+
+        return idle_count == expected_idle_count
+
+    def is_service_running(self, service_name, docker_name):
+        service_status = self.command(
+            "docker exec {} supervisorctl status {}".format(
+                docker_name, service_name
+            ),
+            module_ignore_errors=True
+        )["stdout"]
+
+        logging.info(
+            "{} status:\n%s".format(docker_name),
+            self.command(
+                "docker ps -f name={}".format(docker_name), module_ignore_errors=True
+            )["stdout"]
+        )
```

Comment:
Would it make more sense to move this log up to the `is_container_present` call and add a log here to show the `supervisorctl` output?

Reply:
Sounds fine. I added it here since in your latest commit this was log present after checking the service liveness.

Reply:
Wait, I see what you're saying now. I think moving the `docker ps -f` print above is still a good change, but we should still have a log message for `docker exec ... supervisorctl status` here.

Reply:
sure, I'll add.